test/backend/QueryScheduler.spec.js

```diff
@@ -163,6 +163,219 @@ describe('QueryScheduler', function() {
+        let clock = sinon.useFakeTimers();
+        const ONE_HOUR = 3600;
+
+        const spy1 = sinon.spy();
+        queryScheduler.job = spy1;
+        queryScheduler.scheduleQuery({
+            cronInterval: '31 * * * *', // every hour at minute 31
+            fid: '...',
+            uids: '...',
+            query: '...',
+            connectionId: '...'
+        });
+
+        assert(spy1.notCalled, 'job should not have been called yet');
+        clock.tick(0.5 * ONE_HOUR * 1000);
+        assert(spy1.notCalled, 'job should not have been called yet');
+        clock.tick(ONE_HOUR * 1000);
+        assert(spy1.called, 'job should have been called');
+        clock.tick(ONE_HOUR * 1000);
+        assert(spy1.calledTwice, 'job should have been called twice');
+
+        clock.restore();
+        clock = sinon.useFakeTimers();
+
+        const spy2 = sinon.spy();
+        queryScheduler.job = spy2;
+        queryScheduler.scheduleQuery({
+            cronInterval: '29 * * * *', // every hour at minute 29
+            fid: '...',
+            uids: '...',
+            query: '...',
+            connectionId: '...'
+        });
+
+        assert(spy2.notCalled, 'job should not have been called yet');
+        clock.tick(0.5 * ONE_HOUR * 1000);
+        assert(spy2.called, 'job should have been called');
+        clock.tick(ONE_HOUR * 1000);
+        assert(spy2.calledTwice, 'job should have been called twice');
+
+        clock.restore();
+    });
+
+    it('correctly schedules every day cron interval', function () {
+        const clock = sinon.useFakeTimers();
+        const currentHour = (new Date()).getHours();
+
+        const ONE_MINUTE = 60;
+        const ONE_DAY = 86400;
+
+        const spy = sinon.spy();
+        queryScheduler.job = spy;
+        queryScheduler.scheduleQuery({
+            // every day at 15 minutes past the current hour
+            cronInterval: `15 ${currentHour} * * *`,
+            fid: '...',
+            uids: '...',
+            query: '...',
+            connectionId: '...'
+        });
+
+        assert(spy.notCalled, 'job should not have been called yet');
+        clock.tick(14 * ONE_MINUTE * 1000);
+        assert(spy.notCalled, 'job should not have been called yet');
+        clock.tick(2 * ONE_MINUTE * 1000);
+        assert(spy.called, 'job should have been called');
+        clock.tick(ONE_DAY * 1000);
+        assert(spy.calledTwice, 'job should have been called twice');
+        clock.tick(8 * ONE_DAY * 1000);
+        assert.equal(spy.callCount, 10, 'job should have been called twice');
+
+        clock.restore();
+    });
+
+    it('correctly schedules weekly cron interval', function () {
+        const clock = sinon.useFakeTimers();
+        const currentHour = (new Date()).getHours();
+        const currentDay = (new Date()).getDay();
+
+        const ONE_MINUTE = 60;
+        const ONE_DAY = 86400;
+        const ONE_WEEK = ONE_DAY * 7;
+
+        if (currentDay !== 3) {
```

Comment:
I don't understand the purpose of this `if`. It looks like it was meant to be a `while` to move `currentDay` back to Wednesday.